### PR TITLE
Add abillity to open to specific item within a Jupyter book

### LINKED ISSUE #7069: Add ability to open Jupyter book to specific notebook
Initial request is for linking to specific TSG notebooks from various dashboard pages

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Not used at the moment but is an upcoming requirement for the BDC Dashboard. 

Closes #7069

### CODE REVIEW

File: extensions/notebook/src/book/bookTreeItem.ts [hunk 1]
```diff
@@ -137,4 +137,29 @@ export class BookTreeItem extends vscode.TreeItem {
...
+	/**
+	 * Helper method to find a child md file or Notebook with a specified URL
+	 * @param section The current section we're checking
+	 * @param url The url of the md file or Notebook we're searching for
+	 */
+	public findChildItem(url: string): any {
+		return this.findChildItemRecur(this, url);
+	}
+
+	private findChildItemRecur(section: any, url: string): any {
```
Comment (chlafreniere) [inline extensions/notebook/src/book/bookTreeItem.ts]: Can we return something other than any here?

File: extensions/notebook/src/book/bookTreeItem.ts [hunk 2]
```diff
@@ -137,4 +137,29 @@ export class BookTreeItem extends vscode.TreeItem {
...
+		if (section.url && section.url === url) {
+			return section;
+		}
+		else if (section.sections) {
```
Comment (chlafreniere) [inline extensions/notebook/src/book/bookTreeItem.ts]: nit: else if on previous line

File: extensions/notebook/src/book/bookTreeItem.ts [hunk 3]
```diff
@@ -137,4 +137,29 @@ export class BookTreeItem extends vscode.TreeItem {
...
+			for (let i = 0; i < section.sections.length; i++) {
```
Comment (chlafreniere) [inline extensions/notebook/src/book/bookTreeItem.ts]: is this the preferred method of doing this (speed)? any reason you aren't doing a .forEach (or something else entirely)?

File: extensions/notebook/src/book/bookTreeItem.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/notebook/src/book/bookTreeItem.ts]: Not really. We don't have typing information for Jupyter book TOC. We could make our own types but at the end of the day we still can't rely on the TOC JSON always being "valid" as we know it so all that would get us is dev-time typing and I didn't feel it was worth the effort to do that at this point. 

---
In reply to: [323449823](https://github.com/microsoft/azuredatastudio/pull/7155#discussion_r323449823) [](ancestors = 323449823)

**Pull request merged**